src/torrent.cpp

```diff
@@ -4040,22 +4040,45 @@ namespace {
 				ret = false;
 				break;
 			}
-			auto result = get_hash_picker().set_block_hash(piece, i * default_block_size, block_hashes[i]);
+			auto const result = get_hash_picker().set_block_hash(piece
+				, i * default_block_size, block_hashes[i]);
 
-			block_passed[i] = result.status == set_block_hash_result::result::success;
 			if (result.status == set_block_hash_result::result::success)
 			{
-				TORRENT_ASSERT(result.first_verified_block <= blocks_in_piece);
```

Comment:
my understanding is that `first_verified_block` should never be `== blocks_in_piece`. That would only make sense if `num_verified == 0`. I strengthened this assert by saying it's `< blocks_in_piece`